File: Problem/Kakao/2020_kakao_blind_recruitment_q1.py
INF = int(1e9)


def solution(s):
    answer = INF

    if len(s) == 1:  # 길이가 1일 때, return 1
        return 1

    for cutPoint in range(1, len(s) // 2 + 1):
        ans = ""
        count = 1
        tmpStr = s[:cutPoint]

        for idx in range(cutPoint, len(s), cutPoint):
            if tmpStr == s[idx:idx + cutPoint]:
                count += 1
            else:
                if count == 1:
                    count = ""
                ans += str(count) + tmpStr
                tmpStr = s[idx:idx + cutPoint]
                count = 1

        if count == 1:
            count = ""

        ans += str(count) + tmpStr
        answer = min(answer, len(ans))

    return answer

File: Problem/Kakao/2020_kakao_blind_recruitment_Q1.java
// 문자열 압축
class Solution {
    public int solution(String s) {
        int answer = Integer.MAX_VALUE;
        
        for(int i=1;i<=s.length();i++){
            int cnt = 1;
            String str = "";
            String tmp = "";
            
            for(int j=0;j<s.length()/i;j++){
                if(tmp.equals(s.substring((i*j),(i*j)+i))){
                    cnt++;
                    continue;
                }
                
                if(cnt>1){
                    str += cnt+tmp;
                    cnt = 1;
                }else{
                    str += tmp;
                }
                tmp = s.substring((i*j),(i*j)+i);
            }
            
            if(cnt > 1){
                str += cnt+tmp;
            }else{
                str += tmp;
            }
            
            if(s.length()%i != 0){
                str += s.substring(s.length()-s.length()%i, s.length());
            }
            answer = Math.min(answer, str.length());
        }
        return answer;
    }
}